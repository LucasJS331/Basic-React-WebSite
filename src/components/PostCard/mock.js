export const Mock = [
    {
        title: "teste 1",
        cover: "cover teste 1",
        body: "body teste 3"
    },
    {
        title: "teste 2",
        cover: "cover teste 2",
        body: "body teste 2"
    }
]