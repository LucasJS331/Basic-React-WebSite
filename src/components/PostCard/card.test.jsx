import { render, screen } from "@testing-library/react"
import { PostCard } from "."
import {Mock} from "./mock"

describe("<PostCard />", ()=>{
    it("shoud be in the screen",()=>{
        render(<PostCard posts={Mock} />);

        expect(screen.getByRole("heading", {name: "teste 1"} )).toBeInTheDocument;
        expect(screen.getByAltText("teste 1")).toHaveAttribute("src", "cover teste 1");
        expect(screen.getByText("body teste 3")).toBeInTheDocument();
    })

    it("shoud be match with snapshot", ()=>{
     let {container} =  render(<PostCard posts={Mock} />)

     expect(container.firstChild).toMatchSnapshot("PostCard");

    })

    it("should not render the component", ()=>{
        render(<PostCard posts={[]}/>);

        expect(screen.queryByRole("heading", {name: "teste 1"})).not.toBeInTheDocument()

    })
})
