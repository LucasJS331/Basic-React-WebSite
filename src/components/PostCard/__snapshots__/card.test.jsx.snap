// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> shoud be match with snapshot: PostCard 1`] = `
<div
  class="section-card"
>
  <article
    id="card"
  >
    <img
      alt="teste 1"
      class="card-image"
      src="cover teste 1"
    />
    <div
      class="card-content"
    >
      <h1
        class="card-title"
      >
        teste 1
      </h1>
      <p
        class="card-text"
      >
        body teste 3
      </p>
    </div>
  </article>
  <article
    id="card"
  >
    <img
      alt="teste 2"
      class="card-image"
      src="cover teste 2"
    />
    <div
      class="card-content"
    >
      <h1
        class="card-title"
      >
        teste 2
      </h1>
      <p
        class="card-text"
      >
        body teste 2
      </p>
    </div>
  </article>
</div>
`;
